config/management/operational/src/governance.rs

```diff
@@ -2,11 +2,104 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{json_rpc::JsonRpcClientWrapper, TransactionContext};
+use libra_crypto::ed25519::Ed25519PublicKey;
 use libra_global_constants::LIBRA_ROOT_KEY;
-use libra_management::{config::Config, error::Error, transaction::build_raw_transaction};
-use libra_types::{account_address::AccountAddress, account_config::libra_root_address};
+use libra_management::{
+    config::{Config, ConfigPath},
+    error::Error,
+    transaction::build_raw_transaction,
+};
+use libra_types::{
+    account_address::AccountAddress,
+    account_config::libra_root_address,
+    chain_id::ChainId,
+    transaction::{authenticator::AuthenticationKey, Script},
+};
+use std::path::PathBuf;
 use structopt::StructOpt;
 
+#[derive(Debug, StructOpt)]
+pub struct CreateAccount {
+    #[structopt(flatten)]
+    config: ConfigPath,
+    #[structopt(long)]
+    name: String,
+    #[structopt(long)]
+    path_to_key: PathBuf,
+    #[structopt(long, required_unless = "config")]
+    json_server: Option<String>,
+    #[structopt(long, required_unless("config"))]
+    chain_id: Option<ChainId>,
+}
+
+impl CreateAccount {
+    fn config(&self) -> Result<Config, Error> {
+        self.config.load().map(|c| {
+            c.override_chain_id(self.chain_id)
+                .override_json_server(&self.json_server)
+        })
+    }
+
+    fn key(&self) -> Result<Ed25519PublicKey, Error> {
+        libra_management::read_key_from_file(&self.path_to_key)
+            .map_err(|e| Error::UnableToReadFile(format!("{:?}", self.path_to_key), e))
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct CreateValidator {
+    #[structopt(flatten)]
+    input: CreateAccount,
+}
+
+impl CreateValidator {
+    pub fn execute(self) -> Result<(TransactionContext, AccountAddress), Error> {
+        let config = self.input.config()?;
+        let key = self.input.key()?;
+        let client = JsonRpcClientWrapper::new(config.json_server.clone());
+
+        let seq_num = client.sequence_number(libra_root_address())?;
+        let auth_key = AuthenticationKey::ed25519(&key);
+        let account_address = auth_key.derived_address();
+        let script = transaction_builder::encode_create_validator_account_script(
+            seq_num,
+            account_address,
+            auth_key.prefix().to_vec(),
+            self.input.name.as_bytes().to_vec(),
+        );
+
+        build_and_submit_libra_root_transaction(&config, seq_num, script)
+            .map(|a| (a, account_address))
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub struct CreateValidatorOperator {
+    #[structopt(flatten)]
+    input: CreateAccount,
+}
+
+impl CreateValidatorOperator {
+    pub fn execute(self) -> Result<(TransactionContext, AccountAddress), Error> {
```

Comment:
nit: it seems the only difference between this execute() call and the one for the CreateValidator struct is the script type (encode_create_validator_operator_account_script vs encode_create_validator_account_script). Ideally we should refactor into a common helper (e.g., on CreateAccount) to avoid duplication?

Reply:
fair enough